modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -703,13 +703,32 @@ private void _addListTypeDefinition(ObjectDefinition objectDefinition)
 
 	private void _addObjectDefinitionResources(
 			Set<String> accountEntryRestrictedObjectRelationshipsNames,
-			ObjectAction[] objectActions, long objectDefinitionId,
+			ObjectAction[] objectActions,
+			com.liferay.object.model.ObjectDefinition
+				serviceBuilderObjectDefinition,
 			ObjectLayout[] objectLayouts,
-			ObjectRelationship[] objectRelationships,
+			List<ObjectRelationship> objectRelationships,
 			ObjectValidationRule[] objectValidationRules,
 			ObjectView[] objectViews)
 		throws Exception {
 
+		if (serviceBuilderObjectDefinition.isModifiable() &&
```

Comment:
@marcelabc What do you think about following the approach Pedro did in this PR: https://github.com/liferay-objects/liferay-portal/pull/2565/files#diff-05788445836e16c530386e55062a7c76a71e8ce4d60428c58cee9b01a1b50966R561-R665 ?

It's less complex for now.